src/arkode/arkode_io.c

```diff
@@ -34,6 +35,347 @@
+                             "arkode.no_inactive_root_warn",
+                             "arkode.reset_accumulated_error"};
+  arkBoolSetFn bool_set[] = {ARKodeSetNonlinear,
+                             ARKodeClearStopTime,
+                             ARKodeSetNoInactiveRootWarn,
+                             ARKodeResetAccumulatedError};
+
+  int i, j, retval;
+  for (i = 1; i < argc; i++)
+  {
+    sunbooleantype arg_used = SUNFALSE;
+
+    /* skip command-line arguments that do not begin with "arkode." */
+    if (strncmp(argv[i], "arkode.", 7) != 0) { continue; }
+
+    /* check all "int" command-line options */
+    for (j = 0; j < num_int_keys; j++)
+    {
+      if (CheckAndSetIntArg(ark_mem, &i, argv, int_keys[j],
+                            int_set[j], &arg_used) != ARK_SUCCESS)
+      {
+        return ARK_ILL_INPUT;
+      }
+      if (arg_used) break;
+    }
+    if (arg_used) continue;
+
+    /* check all long int command-line options */
+    for (j = 0; j < num_long_keys; j++)
+    {
+      if (CheckAndSetLongArg(ark_mem, &i, argv, long_keys[j],
+                             long_set[j], &arg_used) != ARK_SUCCESS)
+      {
+        return ARK_ILL_INPUT;
+      }
+      if (arg_used) break;
+    }
+    if (arg_used) continue;
+
+    /* check all real command-line options */
+    for (j = 0; j < num_real_keys; j++)
+    {
+      if (CheckAndSetRealArg(ark_mem, &i, argv, real_keys[j],
+                             real_set[j], &arg_used) != ARK_SUCCESS)
+      {
+        return ARK_ILL_INPUT;
+      }
+      if (arg_used) break;
+    }
+    if (arg_used) continue;
+
+    /* check all bool command-line options */
+    for (j = 0; j < num_bool_keys; j++)
+    {
+      if (CheckAndSetBoolArg(ark_mem, &i, argv, bool_keys[j],
+                             bool_set[j], &arg_used) != ARK_SUCCESS)
+      {
+        return ARK_ILL_INPUT;
+      }
+      if (arg_used) break;
+    }
+    if (arg_used) continue;
+
+    /*** handle all remaining command-line options ***/
+
+    if (strcmp(argv[i], "arkode.interpolant_type") == 0)
+    {
+      i++;
+      retval = 1;
+      if (strcmp(argv[i], "ARK_INTERP_HERMITE") == 0)
+      {
+        retval = ARKodeSetInterpolantType(arkode_mem, ARK_INTERP_HERMITE);
+      }
+      else if (strcmp(argv[i], "ARK_INTERP_LAGRANGE") == 0)
+      {
+        retval = ARKodeSetInterpolantType(arkode_mem, ARK_INTERP_LAGRANGE);
+      }
+      else if (strcmp(argv[i], "ARK_INTERP_NONE") == 0)
+      {
+        retval = ARKodeSetInterpolantType(arkode_mem, ARK_INTERP_NONE);
+      }
+      if (retval != ARK_SUCCESS)
+      {
+        arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                        "error setting command-line argument: %s %s",
+                        argv[i-1], argv[i]);
+        return ARK_ILL_INPUT;
+      }
+      arg_used = SUNTRUE;
+      continue;
+    }
+
+    if (strcmp(argv[i], "arkode.tolerances") == 0)
```

Comment:
Maybe this should be `arkode.scalar_tolerances` so we could have `arkode.vector_tolerances`. Setting the vector tolerance would be more complicated though e.g., if using a distributed vector, GPU vector, user-defined vector, etc. Do we plan on supporting vector-valued inputs from the command line?

Reply:
We do not have to support vector-valued inputs in the first pass, but yes, we plan to support vector-valued. The design doc covers non-distributed vectors. For distributed vectors, we will have to think about it more. Initially, I would be inclined to take the first `local_size` vector-value inputs and copy them to all ranks.

Reply:
It would be nice to have separate arguments for absolute and relative tolerances. However, since the tolerance 'set' functions take both as inputs it would be a little complicated to implement. Maybe we could call the tolerance function twice?

Reply:
> I dont think we need to discuss the distributed aspect here though, we should do it in the design document.

Agreed, we'll need to work through this for GPU and ManyVectors also. I think we should be able to leverage the buffer pack and unpack functions to make these work.

> Maybe we could call the tolerance set function twice?

I think this would be a little tricky since we need to save the values for the later calls